File: Data Structures & Algorithms/is-palindrome/submission-0.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = s.lower()
        newString = ' '.join([char for char in s if char.isalpha()])

        for i in range (0, int(len(newString)/2 + 1)):
            if (newString[i] != newString[len(newString)-i-1]):
                return False
        return True

        

File: Data Structures & Algorithms/valid-palindrome-ii/submission-0.cpp
class Solution {
public:
    bool validPalindrome(string s) {
        int n = 1;
        int p1 = 0, p2 = s.size()-1;
        while(p1<p2)
        {
            if(n<0) return false;
            if(s[p1] == s[p2]){
                p1++;
                p2--;
            }
            else if(s[p1] != s[p2]){
                if(n==0) return false;
                else if((s[p1+1] != s[p2]) && (s[p1] != s[p2-1])){
                    return false;
                }
                else if(s[p1+1] == s[p2]){
                    p1++;
                    n--;
                }
                else if(s[p1] == s[p2-1]){
                    p2--;
                    n--;
                }
            }
        }
        return true;
    }
};